core/domain/stats_services.py

```diff
@@ -16,251 +16,137 @@
 
 """Services for exploration-related statistics."""
 
-import itertools
-
-from core.domain import exp_services
-from core.domain import interaction_registry
+from core.domain import exp_domain
 from core.domain import stats_domain
-from core.domain import stats_jobs_continuous
 from core.platform import models
 
 (stats_models,) = models.Registry.import_models([models.NAMES.statistics])
 
 
-# TODO(bhenning): Test.
-def get_visualizations_info(exploration_id, state_name):
-    """Returns a list of visualization info. Each item in the list is a dict
-    with keys 'data' and 'options'.
+def handle_new_exp_stats_creation(exploration):
+    """Creates ExplorationStatsModel for the freshly created exploration and
+    sets all initial values to zero.
 
     Args:
-        exploration_id: str. The exploration ID.
-        state_name: str. Name of the state.
-
-    Returns:
-        list(dict). Each item in the list is a dict with keys representing
-        - 'id': str. The visualization ID.
-        - 'data': list(dict). A list of calculation IDs.
-        - 'options': dict. The visualization options.
-
-        An example of the returned value may be:
-        [{'options': {'y_axis_label': 'Count', 'x_axis_label': 'Answer'},
-        'id': 'BarChart',
-        'data': [{u'frequency': 1, u'answer': 0}]}]
+        exploration: Exploration. The exploration domain object.
     """
-    exploration = exp_services.get_exploration_by_id(exploration_id)
```

Comment:
Having actually looked at this file, perhaps what you should do instead in order to break the dependency on exp_services is to just pass in the exploration domain object. There are only two calls to exp_services and they are just simple get_exploration_by_id() operations. Then you don't need to do the more complicated solution of having an "old stats service" and a "current stats service" -- I didn't realize, at the time I suggested it, that the dependency was actually pretty small and (I think) easily breakable.

Reply:
That sounds fine by me. I actually thought about that - but didn't want to go around breaking up parameters for functions we might not end up using. But, it seems to be a better alternative to the new service file.

Reply:
done.